### ISSUE TITLE
[use may own pretrain model]

### ISSUE THREAD

**Issue by y-kl8** [zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow#45]
If I want to use may own pretrain resnet50 model, the tensor name in my ckpt file is:
 tensor_name group2/block3/conv3/bn/mean/EMA

tensor_name group3/block1/conv2/bn/variance/EMA

tensor_name group3/block0/conv1/bn/beta

tensor_name group3/block2/conv3/bn/beta/Momentum

tensor_name group3/block2/conv1/bn/gamma

tensor_name group2/block4/conv3/W/Momentum

tensor_name group0/block1/conv3/bn/gamma/Momentum

tensor_name group1/block0/conv3/W
........

and in your ckpt file, the tensor name is:
tensor_name resnet_v1_50/block3/unit_2/bottleneck_v1/conv1/BatchNorm/moving_mean

tensor_name resnet_v1_50/block4/unit_1/bottleneck_v1/conv3/BatchNorm/beta

tensor_name resnet_v1_50/block3/unit_2/bottleneck_v1/conv3/BatchNorm/gamma
tensor_name resnet_v1_50/block2/unit_1/bottleneck_v1/conv3/weights

tensor_name resnet_v1_50/block3/unit_1/bottleneck_v1/conv3/BatchNorm/moving_variance
........
I have changed the name in network.py:

		with tf.variable_scope(scope_name) as scope:
			outputs = self._start_block('conv0')
			print("after start block:", outputs.shape)
			with tf.variable_scope('group0') as scope:
				outputs = self._bottleneck_resblock(outputs, 256, 'block0',	identity_connection=False)
				outputs = self._bottleneck_resblock(outputs, 256, 'block1')
				outputs = self._bottleneck_resblock(outputs, 256, 'block2')
				print("after group0 :", outputs.shape)
..........
 def _bottleneck_resblock(self, x, num_o, name, half_size=False, identity_connection=True):
		first_s = 2 if half_size else 1
		assert num_o % 4 == 0, 'Bottleneck number of output ERROR!'
		# branch1
		if not identity_connection:
			o_b1 = self._conv2d(x, 1, num_o, first_s, name='%s/shortcut' % name)
			o_b1 = self._batch_norm(o_b1, name='%s/shortcut' % name, is_training=False, activation_fn=None)
		else:
			o_b1 = x
		# branch2
		o_b2a = self._conv2d(x, 1, num_o / 4, first_s, name='%s/conv1' % name)
		o_b2a = self._batch_norm(o_b2a, name='%s/conv1' % name, is_training=False, activation_fn=tf.nn.relu)

		o_b2b = self._conv2d(o_b2a, 3, num_o / 4, 1, name='%s/conv2' % name)
		o_b2b = self._batch_norm(o_b2b, name='%s/conv2' % name, is_training=False, activation_fn=tf.nn.relu)

		o_b2c = self._conv2d(o_b2b, 1, num_o, 1, name='%s/conv3' % name)
		o_b2c = self._batch_norm(o_b2c, name='%s/conv3' % name, is_training=False, activation_fn=None)
		# add
		outputs = self._add([o_b1,o_b2c], name='%s/add' % name)
		# relu
		outputs = self._relu(outputs, name='%s/relu' % name)
		return outputs
........

what others I should do to run this code with my own pretrain ckpt file?

**Comment by zhengyang-wang** [comment]
Use [this file](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/tools/inspect_checkpoint.py) to check all variables' names in your ckpt file and make sure you generate a graph with the same name.
